src/plugins/filedelete/filedelete2.cpp

```diff
@@ -714,6 +713,57 @@ static event_response_t writefile_cb(drakvuf_t drakvuf, drakvuf_trap_info_t* inf
     return 0;
 }
 
+static event_response_t createsection_cb(drakvuf_t drakvuf, drakvuf_trap_info_t* info)
+{
+    filedelete* f = (filedelete*)info->trap->data;
+    vmi_instance_t vmi = drakvuf_lock_and_get_vmi(drakvuf);
+
+    unicode_string_t* filename_us = nullptr;
+    std::string filename = "<UNKNOWN>";
+    handle_t handle = 0;
+    uint32_t access_mask = 0;
+    access_context_t ctx =
+    {
+        .translate_mechanism = VMI_TM_PROCESS_DTB,
+        .dtb = info->regs->cr3,
+    };
+
+    if (f->pm == VMI_PM_IA32E)
+    {
+        access_mask = (uint32_t) info->regs->rdx;
+        ctx.addr = info->regs->rsp + 7 * sizeof(uint64_t);
+        if ( VMI_FAILURE == vmi_read_64(vmi, &ctx, (uint64_t*) &handle) )
+            goto done;
+        // Filter out system handles: those having high bits rised
+        // WARNING Without this target VM could freeze or crash!
+        if (handle & ~(0x7fffffffULL))
```

Comment:
So this is easier to filter by casting handle to be a signed integer and then just checking if "handle < 0"

Reply:
This is sort of mask. I think it more clear what I check here.

Reply:
No, I think it's more clear what you check for here if its a negative integer check and not the inverse of a magic hex value. That's effectively what this is.

Reply:
@tklengyel, Done!